### ISSUE TITLE
Horizontal, row orientated box plot

### ISSUE THREAD

**Issue by atomless** [dc-js/dc.js#1310]
Any idea what would be involved in generating a box plot orientated along the x axis instead of the y axis? I'm presuming this would be a whole new chart like row chart is separate to bar chart even though they are basically the same thing?

**Comment by gordonwoodhull** [comment]
Yeah, it's unfortunate that the coordinate grid mixin doesn't support vertical orientation - it means there is a certain amount of duplication in the `rowChart` code.

One complication with this feature in particular is that the [d3-plugins/box.js](https://github.com/d3/d3-plugins/blob/master/box/box.js) code this is based on is pretty specific about its Xs and Ys. One might have to transform the view. :-/

**Comment by atomless** [comment]
Okay. Thanks @gordonwoodhull, that's pretty much what I thought. Will have to write our own along the lines of the existing boxplot and see if we can contribute it back to the community.

**Comment by gordonwoodhull** [comment]
It would probably not be difficult to fork box.js and generalize the orientation. My understanding is that all of d3-plugins is deprecated and not supported, so it's not like you would be stepping on any toes. 

It's a few years old and there may be other d3-based libraries out there by now.

**Comment by atomless** [comment]
Interesting - is there some other charting lib I should be looking at other than dc.js? Specifically for crossfilter style charts where interaction/filterin gon one updates related charts on the page.

**Comment by gordonwoodhull** [comment]
Oh, I just meant there are probably other box plot implementations out there. 

I don't know of any alternatives to dc.js. Not that it's all that complicated or that you couldn't build the functionality into other charting libraries, but to my knowledge no other library has this focus.

**Comment by atomless** [comment]
@gordonwoodhull Just looking into this again. So far I'm thinking a new version of `d3.box` called `d3.boxHoizontal` with basically the same code but with x/y flipped. Then a new version of `dc.boxPlot` called `dc.boxPlotHorizontal` altered in much the same way. Also have that use a new version of coordinate grig mixin. I've been wondering about just making coordinate grid mixin able to support both orientations but thought it better to start with separate code and possibly tackle that later. Am I barking up any wrong trees here? Any guidance as to best or your prefered approach would be very much appreciated.

**Comment by gordonwoodhull** [comment]
That would be a great start. Of course, it would be cleaner to generalize  `d3.box` to handle the horizontal case, but sometimes it's easier to fork first, figure out what's needed, and then merge later if possible.

Fixing the `coordinateGridMixin` is probably a lot more work. IIRC a couple of people have tried this, but they determined that the differences in text placement and layout make this more complicated than it sounds. (If you compare the bar chart and the row chart in detail, you'll notice many differences, which are all intentional.)

**Comment by atomless** [comment]
Okay, so long as I'm not barkin gup the wrong trees. Let's see where I get to. Thanks!
